### DISCUSSION TITLE
We're installing using the package and have some different errors

### DISCUSSION THREAD

**Discussion by smr1619** [grahampugh/erase-install#321] [Q&A]
For the most part all the devices I've used the erase-install on has worked great. We're having some test users run this from self-service, but they're getting different errors or they say it never restarts. I'm assuming they're not letting it sit to automatically install to finish the process. Some are downloading to the /Applications folder, but not kicking off.

Preparing: 31.0%            Failed to download required installer asset (SFR).

   [erase-install] attempting to terminate the 'caffeinate' process - Termination message indicates success
/Library/Management/erase-install/erase-install.sh: line 1078:  8217 Terminated: 15          /usr/bin/caffeinate -dimsu -w $$

/Library/Management/erase-install/erase-install.sh: line 795: 10695 Killed: 9               dep_notify_progress startosinstall > /dev/null 2>&1

2nd Error
13  012-90254       12.6.1     21G217   2022-10-24  macOS Monterey                 
14  012-90253       11.7.1     20G918   2022-10-24  macOS Big Sur                  
15  012-93766       13.0.1     22A400   2022-11-09  macOS Ventura                  

Build 21G217 selected. Downloading #13...

Traceback (most recent call last):
  File "/Library/Management/erase-install/installinstallmacos.py", line 1367, in <module>
    main()
  File "/Library/Management/erase-install/installinstallmacos.py", line 1305, in main
    os.link(download_pkg, local_pkg)
FileExistsError: [Errno 17] File exists: '/Library/Management/erase-install/content/downloads/36/13/012-90254-A_BJQ1VMPD44/z225i7bzise31eo21e4kgfkafz7zq0q9tu/InstallAssistant.pkg' -> '/Library/Management/erase-install/InstallAssistant-12.6.1-21G217.pkg'
   [run_installinstallmacos] Error obtaining valid installer. Cannot continue.

   [erase-install] attempting to terminate the 'caffeinate' process - Termination message indicates success
/Library/Management/erase-install/erase-install.sh: line 1078: 12924 Terminated: 15          /usr/bin/caffeinate -dimsu -w $$

/Library/Management/erase-install/erase-install.sh: line 795: 13000 Killed: 9               dep_notify_progress installinstallmacos > /dev/null 2>&1

Edit
To add another issue I just came across. When I first tested this script it was a few days prior to Ventura coming out. It worked my test devices, but apparently I had it set to download the latest OS and didn't realize Ventura just came out until I ran it on my test Mac and it installed Ventura. I had this set in self service for my team to test. One engineer ran it, but it didn't complete the install. That was a few weeks ago, so after multiple tests, I had him run it again, but it was trying to install Ventura on his Mac. Could it have cached the installer in the erase-install folder? Since then, I have used this on 9 devices and it's worked.

log/Library/Management/erase-install/erase-install.sh --reinstall --pkg --os=12 --update --min-drive-space=35 --current-user --check-power --no-fs --depnotify --cleanup-after-use

**Comment by grahampugh** [comment]
It's always somewhat more guess work without the full output, however, here's my guesses:

Error 1 is a network error/interruption.

Error 2 there is something preventing the existing files in the cache being overwritten - perhaps security software? You might need to delete the working directory (/Library/Management/erase-install) on that client and try again.

**Comment by grahampugh** [comment]
To answer your third issue, yes the Ventura installer could have been cached.

**Comment by smr1619** [comment]
[erase-install] Caffeinating this script (pid=44755)
   [check_free_space] OK - 418 GB free/purgeable disk space detected
   [check_power_status] OK - AC power detected
   [erase-install] Looking for existing installer app or pkg
   [find_existing_installer] No valid installer found.
   [erase-install] Running on architecture arm64
yes Michael.Mikoly is a member of everyone
   [get_user_details] User is a Volume Owner
   [get_user_details] ask for password (attempt 1/5)
   [check_password] Success: the password entered is the correct login password for User.
   [erase-install] Opening DEPNotify download message (language=en)
   [get_installinstallmacos] installinstallmacos.py is in /Library/Management/erase-install
   [get_relocatable_python] Relocatable Python is installed in /Library/Management/erase-install
   [run_installinstallmacos] Checking that selected version 12.6.1 is available

   [run_installinstallmacos] This command is now being run:

   installinstallmacos.py --workdir /Library/Management/erase-install --ignore-cache --pkg --version 12.6.1 --validate
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0 11.5G    0  115k    0     0   350k      0  9:35:00 --:--:--  9:35:00  371k
  0 11.5G    0 8511k    0     0  6399k      0  0:31:32  0:00:01  0:31:31 6487k
 
99 11.5G   99 11.4G    0     0  8182k      0  0:24:39  0:24:32  0:00:07 8126k
99 11.5G   99 11.4G    0     0  8182k      0  0:24:39  0:24:33  0:00:06 8121k
99 11.5G   99 11.5G    0     0  8182k      0  0:24:39  0:24:34  0:00:05 8124k
99 11.5G   99 11.5G    0     0  8183k      0  0:24:39  0:24:35  0:00:04 8123k
99 11.5G   99 11.5G    0     0  8183k      0  0:24:39  0:24:36  0:00:03 8572k
99 11.5G   99 11.5G    0     0  8183k      0  0:24:39  0:24:37  0:00:02 8569k
99 11.5G   99 11.5G    0     0  8183k      0  0:24:39  0:24:38  0:00:01 8572k
99 11.5G   99 11.5G    0     0  8184k      0  0:24:39  0:24:39 --:--:-- 8572k
100 11.5G  100 11.5G    0     0  8184k      0  0:24:39  0:24:39 --:--:-- 8573k

installinstallmacos.py - get macOS installers from the Apple software catalog

This Mac:
Model Identifier : MacBookPro18,1

Device ID        : J316sAP
OS Version       : 12.5.1
Build ID         : 21G83

 #  ProductID       Version    Build    Post Date   Title                          
 1  071-78704       11.5.2     20G95    2021-08-18  macOS Big Sur                  
 2  002-23589       11.6.1     20G224   2021-12-01  macOS Big Sur                  
 3  002-42341       11.6.2     20G314   2022-01-14  macOS Big Sur                  
 4  002-57023       11.6.3     20G415   2022-01-26  macOS Big Sur                  
 5  002-65695       11.6.4     20G417   2022-02-17  macOS Big Sur                  
 6  002-77154       11.6.5     20G527   2022-04-11  macOS Big Sur                  
 7  012-08272       11.6.6     20G624   2022-05-24  macOS Big Sur                  
 8  012-40387       11.6.8     20G730   2022-07-28  macOS Big Sur                  
 9  012-51693       12.5.1     21G83    2022-08-24  macOS Monterey                 
10  012-40494       12.6       21G115   2022-09-20  macOS Monterey                 
11  012-38280       11.7       20G817   2022-09-20  macOS Big Sur                  
12  012-92138       13.0       22A380   2022-10-24  macOS Ventura                  
13  012-90254       12.6.1     21G217   2022-10-24  macOS Monterey                 
14  012-90253       11.7.1     20G918   2022-10-24  macOS Big Sur                  
15  012-93766       13.0.1     22A400   2022-11-09  macOS Ventura                  

Build 21G217 selected. Downloading #13...

Package downloaded to: /Library/Management/erase-install/InstallAssistant-12.6.1-21G217.pkg
   [run_installinstallmacos] InstallAssistant package downloaded to /Library/Management/erase-install/InstallAssistant-12.6.1-21G217.pkg.
   [erase-install] Closing DEPNotify download message (language=en)
/Library/Management/erase-install/erase-install.sh: line 795: 47320 Killed: 9               dep_notify_progress installinstallmacos > /dev/null 2>&1
   [check_free_space] OK - 405 GB free/purgeable disk space detected

   [unpack_pkg_to_applications_folder] Unpacking /Library/Management/erase-install/InstallAssistant-12.6.1-21G217.pkg into /Applications folder
installer: Package name is InstallAssistant-12.6.1-21G217
installer: Upgrading at base path /
installer: The upgrade was successful.
   [unpack_pkg_to_applications_folder] Deleting /Library/Management/erase-install/InstallAssistant-12.6.1-21G217.pkg
   [erase-install] WARNING! Running /Applications/Install macOS Monterey.app with reinstall option

   [erase-install] Opening DEPNotify message (language=en)
   [erase-install] Writing LaunchDaemon which will remove /Library/Management/erase-install at next boot
By using the agreetolicense option, you are agreeing that you have run this tool with the license only option and have read and agreed to the terms.
If you do not agree, press CTRL-C and cancel this process immediately.
Preparing to run macOS Installer...
Preparing: 0.0%           
Preparing: 0.1%           
Preparing: 0.2%           
Preparing: 2.4%           
Preparing: 2.5%           
Preparing: 2.6%           
Preparing: 2.7%           
Preparing: 2.8%           
Preparing: 2.9%           
Preparing: 3.0%           
Preparing: 3.1%           
Preparing: 3.2%           
Preparing: 3.3%           
Preparing: 3.4%           
Preparing: 3.5%           
Preparing: 3.6%           
Preparing: 3.7%           
Preparing: 3.8%           
Preparing: 3.9%           
Preparing: 4.0%           
Preparing: 4.1%           
Preparing: 4.2%           
Preparing: 4.3%           
Preparing: 4.4%           
Preparing: 4.5%           
Preparing: 4.6%           
Preparing: 4.7%            
Preparing: 4.8%           
Preparing: 4.9%           
Preparing: 5.0%           
Preparing: 5.1%           
Preparing: 5.2%           
Preparing: 5.3%           
Preparing: 5.4%           
Preparing: 5.5%           
Preparing: 5.6%            
Preparing: 5.7%           
Preparing: 5.8%           
Preparing: 5.9%           
Preparing: 6.0%           
Preparing: 6.1%           
Preparing: 6.2%           
Preparing: 6.3%           
Preparing: 6.4%           
Preparing: 6.5%            
Preparing: 6.6%           
Preparing: 6.7%           
Preparing: 6.8%           
Preparing: 6.9%           
Preparing: 7.0%           
Preparing: 7.1%           
Preparing: 7.2%           
Preparing: 7.3%           
Preparing: 7.4%           
Preparing: 7.5%           
Preparing: 7.6%           
Preparing: 7.7%           
Preparing: 7.8%           
Preparing: 7.9%           
Preparing: 8.0%           
Preparing: 8.1%           
Preparing: 8.2%           
Preparing: 8.3%           
Preparing: 8.4%           
Preparing: 8.5%           
Preparing: 8.6%           
Preparing: 8.7%           
Preparing: 8.8%           
Preparing: 8.9%           
Preparing: 9.0%           
Preparing: 9.1%           
Preparing: 9.2%           
Preparing: 9.5%           
Preparing: 13.7%           
Preparing: 13.9%           
Preparing: 14.5%           
Preparing: 14.6%           
Preparing: 15.4%           
Preparing: 31.0%            Failed to download required installer asset (SFR).

   [erase-install] attempting to terminate the 'caffeinate' process - Termination message indicates success
/Library/Management/erase-install/erase-install.sh: line 1078: 44777 Terminated: 15          /usr/bin/caffeinate -dimsu -w $$


/Library/Management/erase-install/erase-install.sh: line 795: 51667 Killed: 9               dep_notify_progress startosinstall > /dev/null 2>&1

**Comment by smr1619** [comment]
Have another error now on my primary Mac where this has worked before. We are using a jamf helper that gives them a dialogue at first. It's downloading it, but when it hits 1% on the Upgrade screen it goes away. I've tried this on my home network which has worked before and in our Office where it's worked. Below is the command I'm usi
[Erase_Error.txt](https://github.com/grahampugh/erase-install/files/10414408/Erase_Error.txt)
ng.

/Library/Management/erase-install/erase-install.sh --reinstall --pkg --version=12.6.2 --min-drive-space=35 --current-user --fs --depnotify --cleanup-after-use

**Reply by grahampugh** [reply]
Not sure if you read your log output, but it's pretty clear:

```
Preparing: 32.6%            Failed to download required installer asset (SFR).
```

Your preparation failed. From that output (which comes from startosinstall, not from the erase-install script), it looks like more things are being downloaded during the preparation phase, but that connection failed.

PS you said you're using jamfHelper but you're using DEPNotify.

**Reply by smr1619** [reply]
Thanks for the quick reply on this. Thanks for the info, I'll  look into this issue, because I'm not sure what else would trying to download. It's downloading the the app, just not installing it. For the jamfhelper I should have said script, we were testing a script with some info regarding the update and for them to kick it of.  Thanks again!
